tools/chapel-py/src/chapel/visitor/__init__.py

```diff
@@ -0,0 +1,90 @@
+import chapel.core
+
+from collections import defaultdict
+from inspect import signature
+
+def _try_call(dispatch_table, visitor, node, method_type):
+    node_type = type(node)
+    while node_type is not object:
+        if node_type in dispatch_table:
+            methods = dispatch_table[node_type]
+            if method_type in methods:
+                return [methods[method_type](visitor, node)]
```

Comment:
Why return a `List[bool]` and not just a bool?

Reply:
I think this is so `enter` methods can return None (which is the default) and be treated as true. If so, I'd suggest just returning booleans, and then on the check for enter returning true add a case for None

Reply:
I like it because it distinguishes the return type, whatever it is, from there not being a matching function. In my opinion, this is a useful API to have.

Reply:
Right, but that makes more sense as `None`, `True`, `False`. Returning `[False]` or `[True]` is confusing (especially without a comment). I had to reread the code several times to understand why you wrote it that way.

I would prefer just plain `None`, `True`, or `False`, but I do understand your reasoning here. At the very least, you should document this well in the code with comments (which it looks like you have already done)

Reply:
Having read the fresh comment, this almost implies that you should introduce an Enum to handle all of the cases that you want to return, instead of using the python type system in this way

Reply:
Resolved offline 😄